binfmt/binfmt_execmodule.c

```diff
@@ -164,23 +164,23 @@ int exec_module(FAR const struct binary_s *binp,
 #if defined(CONFIG_ARCH_ADDRENV) && defined(CONFIG_BUILD_KERNEL)
   /* Instantiate the address environment containing the user heap */
 
-  ret = up_addrenv_select(&binp->addrenv, &oldenv);
+  ret = addrenv_select((struct addrenv_s *)&binp->addrenv);
   if (ret < 0)
     {
-      berr("ERROR: up_addrenv_select() failed: %d\n", ret);
+      berr("ERROR: addrenv_select() failed: %d\n", ret);
       goto errout_with_envp;
     }
 
-  ret = up_addrenv_vheap(&binp->addrenv, &vheap);
+  ret = up_addrenv_vheap(&binp->addrenv.pgdir, &vheap);
```

Comment:
can we add an local variable to avoid binp->addrenv.pgdir dup four times?

Reply:
Done